source/grid/tria.cc

```diff
@@ -11871,11 +11871,8 @@ Triangulation<dim, spacedim>::create_triangulation(
 
           // boundary ids
           for (auto pair : cell_info->boundary_ids)
-            {
-              Assert(cell->at_boundary(pair.first),
-                     ExcMessage("Cell face is not on the boundary!"));
+            if (cell->face(pair.first)->at_boundary())
```

Comment:
Is this safe? Or do we have a way to check the previous check when we know the data is comprehensive?

Reply:
This is safe. We are re-partitioning `p:f:T`. Here we still don't have a good mean to distinguish between actual boundary cells and calls at the boundary of the locally-relevant coarse grid. Such internal boundaries, cause the problem here - but can be safely ignored.